### PR TITLE
Feature/commands panel

### PR DESCRIPTION
**Pull request by vallq**
Changes:
- Add Commands Panel to the Main Menu
- Shift loop request form from Robots to Commands panel
- Include e2e test for the loop request form*

*Issue: magni robot does not move (on both gazebo and dashboard) even after loop request is sent and trajectory path is charted; causing loopRequest.test.ts to fail intermittently

### CODE REVIEW

File: e2e/tests/loopRequest.test.ts [hunk 1]
```diff
@@ -12,17 +12,31 @@ describe('loop request', () => {
...
-  });
 
-  it('renders robot trajectory', () => {
-    expect($('[data-component=RobotTrajectory]')).toBeVisible();
+    browser.addLocatorStrategy('findAllRobots', () => {
+      return document.querySelectorAll('[data-component=RobotItem]');
+    });
+    const allRobotItems = browser.custom$$('findAllRobots', '[data-component=RobotItem]');
+
+    let getRobotLocations = () => {
```
Comment (koonpeng) [inline e2e/tests/loopRequest.test.ts:21]:
```suggestion
    const getRobotLocations = () => {
      const allRobotItems = browser.custom$$('findAllRobots', '[data-component=RobotItem]');
```

wdio complains of stale elements because `allRobotItems` is resolved, then switched to the commands panel, then switched back to the robots panel and used again. The test still works, I don't know what complications it brings but let's avoid that warning to be safe.

**Comment by vallq** [comment]
**Changes:**
- Updating `loopRequest` parameters
- Loop Request form to make use of TextField error and helperText props
- Updating `loop-firm.test.tsx`:
- [x] initial values are rendered correctly 
- [x] location is not empty
- [x] start and end location are not the same
- [x] submission takes place when form is valid
- [x] submission does not take place when form is invalid
- [x] an error message is present when invalid form is attempted to be submitted

File: src/components/tests/loop-form.test.tsx [hunk 2]
```diff
@@ -15,59 +14,58 @@ const buildWrapper = (listOfPlaces?: string[]) => {
...
-    ).toBeTruthy;
+    expect(wrapper.findWhere(x => x.name() === 'input' && x.props().value != '')).toBeTruthy();
+    expect(wrapper.find('input[name="startLocation"]').props().value).not.toEqual(
+      wrapper.find('input[name="finishLocation"]').props().value,
+    );
     wrapper.unmount();
   });
 
+  test('Successful Request', () => {
+    const spy = jest.spyOn(console, 'log');
```
Comment (koonpeng) [inline src/components/tests/loop-form.test.tsx:37]: This seems like a questionable spy. `console.log` can be called pretty much anywhere.

File: src/components/tests/loop-form.test.tsx [hunk 3]
```diff
@@ -15,59 +14,58 @@ const buildWrapper = (listOfPlaces?: string[]) => {
...
+    const wrapper = buildWrapper('SuperFleet');
+    wrapper.find("input[type='number']").simulate('change', { target: { value: 1 } });
+    expect(wrapper.find("input[type='number']").props().value).toEqual(1);
+    wrapper.find('form').simulate('submit');
+    expect(spy).toBeCalled();
+  });
+
   test('Number of loops cannot be empty', async () => {
-    const wrapper = buildWrapper();
+    const spy = jest.spyOn(console, 'log');
```
Comment (koonpeng) [inline src/components/tests/loop-form.test.tsx:46]: This seems like a questionable spy. console.log can be called pretty much anywhere.

File: src/components/tests/loop-form.test.tsx [hunk 4]
```diff
@@ -15,59 +14,58 @@ const buildWrapper = (listOfPlaces?: string[]) => {
...
     wrapper.find('form').simulate('submit');
-    expect(wrapper.find('#numLoopsError').html()).toContain('Loops can only be &gt; 0');
+    expect(spy).not.toBeCalled();
+    expect(wrapper.exists('.Mui-error')).toBeTruthy();
     wrapper.unmount();
   });
 
   test('Location cannot be empty', async () => {
-    const wrapper = buildWrapper(['placeA']);
+    const spy = jest.spyOn(console, 'log');
```
Comment (koonpeng) [inline src/components/tests/loop-form.test.tsx:55]: This seems like a questionable spy. console.log can be called pretty much anywhere.

File: src/components/tests/loop-form.test.tsx [hunk 5]
```diff
@@ -15,59 +14,58 @@ const buildWrapper = (listOfPlaces?: string[]) => {
...
-    expect(wrapper.find('#startLocationError').html()).toContain('Location cannot be empty');
+    expect(spy).not.toBeCalled();
+    expect(wrapper.exists('.Mui-error')).toBeTruthy();
     wrapper.unmount();
   });
 
-  test('Start Location cannot be equal to finish Location', async () => {
-    const wrapper = buildWrapper(['placeA', 'placeA', 'placeA']);
+  test('Start Location cannot be equal to Finish Location', async () => {
+    const spy = jest.spyOn(console, 'log');
```
Comment (koonpeng) [inline src/components/tests/loop-form.test.tsx:64]: This seems like a questionable spy. console.log can be called pretty much anywhere.

File: src/components/loop-form.tsx [hunk 6]
```diff
@@ -93,46 +126,46 @@ export const RobotLoopForm = (props: robotLoopFormProps) => {
...
-          </p>
-        )}
       </div>
 
       <div className={classes.divForm}>
         <Autocomplete
           getOptionLabel={option => option}
           onChange={(e, value) => setStartLocation(value || '')}
-          options={listOfPlaces}
+          options={fakePlaces()[targetFleetName]}
```
Comment (matiasbavera) [inline src/components/loop-form.tsx:137]: Maybe it's a good idea for this to be a `state` or a `Memo`, because in the near future we need to replace this for the real value obtained by a service.

E.g:
```
const listOfPlaces = React.useMemo(()=>fakePlaces()[targetFleetName] )
or 
const [listOfPlaces, setListOfPlaces] = useState([])

useEffect(()=>{
    // When we have the service we will need to modify just this function
    setListOfPlaces.set(fakePlaces()[targetFleetName])
}, [targetFleetName])

options={listOfPlaces}
```

Let me know what you think about it

File: src/components/loop-form.tsx [hunk 7]
```diff
@@ -1,44 +1,52 @@
...
+import { makeStyles, TextField, Typography, Button } from '@material-ui/core';
 import Autocomplete from '@material-ui/lab/Autocomplete';
 import React, { useState } from 'react';
 import { successMsg } from '../util/alerts';
+import fakePlaces from '../mock/data/places';
 
-interface robotLoopFormProps {
-  fleetName: string;
+interface LoopFormProps {
+  fleets: string[];
```
Comment (matiasbavera) [inline src/components/loop-form.tsx:8]: What do you think about adding more descriptive names? it makes the difference when reading the code. E.g if I see without context that you're iterating fleets, I'll think that you're iterating an array of objects with the full content of the fleets and no an array of strings with the name of the fleets. I hope I made my point with this, but let me know if you have any questions.

what do you think about fleetNames, fleetDescriptions? 

I'm not good in assigning names either =)

File: src/components/loop-form.tsx [hunk 8]
```diff
@@ -93,46 +126,46 @@ export const RobotLoopForm = (props: robotLoopFormProps) => {
...
-          </p>
-        )}
       </div>
 
       <div className={classes.divForm}>
         <Autocomplete
           getOptionLabel={option => option}
           onChange={(e, value) => setFinishLocation(value || '')}
-          options={listOfPlaces}
+          options={fakePlaces()[targetFleetName]}
```
Comment (matiasbavera) [inline src/components/loop-form.tsx:156]: Same as line 137

File: src/components/commands-panel.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    startLocationPoint: string,
+    endLocationPoint: string,
+  ) => {
+    requestLoop(loopRequestPub, fleetName, numLoops, startLocationPoint, endLocationPoint);
+  };
+
+  return (
+    <React.Fragment>
+      <LoopForm requestLoop={handleRequestLoop} fleets={allFleets} />
+    </React.Fragment>
```
Comment (matiasbavera) [inline src/components/commands-panel.tsx:48]: Hey @koonpeng would be good to have also an accordion component like this?
image
We will need to add the Delivery Request here too and I think we will add more commands in the future too.

File: src/components/main-menu.tsx [hunk 10]
```diff
@@ -39,6 +40,12 @@ export default function MainMenu(props: MainMenuProps): React.ReactElement {
       <ListItem data-item="Dispensers" button={true} onClick={props.onDispensersClick}>
         <Typography variant="h5">Dispensers</Typography>
       </ListItem>
+
+      <Divider />
+
```
Comment (matiasbavera) [inline src/components/main-menu.tsx:45]: nit: remove extra space?

File: src/components/tests/loop-form.test.tsx (hunk 5 above)
Comment (vallq) [reply src/components/tests/loop-form.test.tsx:64]: @koonpeng Do you have any suggestion regarding this?
what I can think of is using `expect(spy).toBeCalledWith('test')` / `expect(spy).not.toBeCalledWith('test')` and the onClick event handler could produce a more specific console.log statement like 'testing loop form'
```
const onClick = (
    fleetName: string,
    numLoops: number,
    startLocationPoint: string,
    endLocationPoint: string,
  ) => {
    console.log('testing the loop form');
  };
...
expect(spy).toBeCalledWith('testing the loop form');
```
@matiasbavera if you have any suggestion for this, please do chime in! (:

File: src/components/loop-form.tsx (hunk 6 above)
Comment (vallq) [reply src/components/loop-form.tsx:137]: I was struggling with this a lot actually! so previously `listOfPlaces` was passed as a string array into the LoopForm because the RobotItem would have created `listOfPlaces` with `const listOfPlaces = fakePlaces()[fleetName]` so I was not sure how to keep the listOfPlaces variable when moving the loop form out of the robots panel.
but what you've suggested seems viable, and I agree that it should be easily changed when the real value from the service is made use of.

File: src/components/loop-form.tsx (hunk 7 above)
Comment (vallq) [reply src/components/loop-form.tsx:8]: I understand! fleetNames would be a better option here since it's really just an array of the names, and not the full fleet object. will make the change!

File: src/components/commands-panel.tsx (hunk 9 above)
Comment (vallq) [reply src/components/commands-panel.tsx:48]: So if I understand correctly, each request type will have its own accordion component, right? should this be made in this PR or created in another PR when delivery request is added in?

File: src/components/loop-form.tsx [hunk 11]
```diff
@@ -1,44 +1,52 @@
...
+  const [targetFleetName, setTargetFleetName] = useState(fleets.length >= 1 ? fleets[0] : '');
   const [numLoops, setNumLoops] = useState(0);
   const [startLocation, setStartLocation] = useState(
-    listOfPlaces.length >= 2 ? listOfPlaces[0] : '',
+    fakePlaces()[targetFleetName].length >= 2 ? fakePlaces()[targetFleetName][0] : '',
   );
   const [finishLocation, setFinishLocation] = useState(
-    listOfPlaces.length >= 2 ? listOfPlaces[1] : '',
+    fakePlaces()[targetFleetName].length >= 2 ? fakePlaces()[targetFleetName][1] : '',
   );
```
[the remark this answers is not in the document]
Comment (vallq) [reply src/components/loop-form.tsx:28]: oh I believe this happens because Fleet1 and Fleet2 do not have any places listed under `places.ts` 
image
would adding fleet1 and 2 in be considered good practice or there is another way to do this?

[the remark this answers is not in the document]
Comment (koonpeng) [reply src/components/loop-form.tsx:28]: Yea, just add it for now, this whole `fakePlaces` is a hacky workaround to an ongoing problem here #54

File: src/components/commands-panel.tsx (hunk 9 above)
Comment (koonpeng) [reply src/components/commands-panel.tsx:48]: I thought about this before, that would be nice especially as we add more commands.

File: src/components/tests/loop-form.test.tsx (hunk 5 above)
Comment (matiasbavera) [reply src/components/tests/loop-form.test.tsx:64]: How about something like this?

``` typescript
test('Start Location cannot be equal to Finish Location', async () => {
  let clicked = false;
  const onClick = (
    fleetName: string,
    numLoops: number,
    startLocationPoint: string,
    endLocationPoint: string,
  ) => {
    clicked = true;
  };
  // You can send a custom onClick function to your buildWrapper function
  const wrapper = buildWrapper('FleetB', onClick);
  wrapper.find('form').simulate('submit');
  expect(clicked).toBeFalsy();
  /**
   * Also it should be good to check for the exact error. You can do that setting an Id to the Autocomplete component and accessing to that html element in this way .find('#<ID>-helper-text')
   * You can check an example here https://github.com/osrf/romi-dashboard/blob/2d22429068d014bc7257b0a28cbe8d0fa2c93498/src/components/tests/robot-item-delivery-form.test.tsx
   */
  expect(wrapper.exists('.Mui-error')).toBeTruthy();
  wrapper.unmount();
});

```

File: src/components/loop-form.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (matiasbavera) [reply src/components/loop-form.tsx:28]: Yeah, It's ok, but It'll be good to add a safeguard to the form I the meanwhile. Also, you can add an error saying something like
 * There are no places available for this fleet. Pick another fleet. or
 * This fleet is not enabled to go anywhere. Pick another fleet

You can take this as an example https://github.com/osrf/romi-dashboard/blob/2d22429068d014bc7257b0a28cbe8d0fa2c93498/src/components/robot-item-delivery-form.tsx

File: src/components/commands-panel.tsx (hunk 9 above)
Comment (matiasbavera) [reply src/components/commands-panel.tsx:48]: > So if I understand correctly, each request type will have its own accordion component, right? should this be made in this PR or created in another PR when delivery request is added in?

Yeah, that's right, it should go like this on accordion opened

```
commands 
- command 1
--- Info A1
--- Info A2
- command 2
--- Info B1
--- Info B2
```

And like this on accordion closed

```
commands 
- command 1
- command 2
```

I'll need that for the delivery request PR. I can also make those changes on the Delivery Request PR, but however you want. Let me know.

Comment (vallq) [reply src/components/commands-panel.tsx:48]: Lets go with your suggestion then! And you can make the changes to the Delivery Request PR as well. Will get this done by today

**Comment by vallq** [comment]
Changes:

- [x] Adding accordion / expansion panel component for loop request
- [x] Updating variable names
- [x] Using `listOfPlaces` as state component
- [x] Adding fleet details to mock data
- [x] Adding helper text for fleets that are not available for requests
- [x] Updating tests to not use spy

File: .storybook/main.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  stories: ['../src/**/*.stories.js'],
+  addons: [
+    '@storybook/preset-create-react-app',
+    '@storybook/addon-actions',
+    '@storybook/addon-links',
+  ],
+};
```
Comment (matiasbavera) [inline .storybook/main.js:8]: QQ: did you rebased or merged with master? 

I'm asking this because I think these are old changes that are already in master, so in theory, we don't need to review it again because it already had their review process. In this PR we have 39 files changes that are your changes + old changes (that are already in master). Having this makes a bit difficult to make a proper review of the functionalities that you try to add to the project. With this format, the reviewer would have extra work (to go by each commit to check what are truly your changes and review them).

I prefer to work with rebase, for this kind of situation (but this is just my bias opinion). I hope I'm not annoying and I hope this helps.

Let's check with @koonpeng .

Adding some helpful links:
https://www.atlassian.com/git/tutorials/merging-vs-rebasing
https://medium.com/datadriveninvestor/git-rebase-vs-merge-cc5199edd77c#:~:text=When%20you%20do%20rebase%20a,feature%20branch%20history%20remains%20same.

**Comment by matiasbavera** [comment]
There are two `commands` in the panel, maybe it's for a merge problem
image

**Comment by matiasbavera** [comment]
Thanks for this @vallq 
image

File: .storybook/main.js (hunk 12 above)
Comment (vallq) [reply .storybook/main.js:8]: I used `git pull --rebase` on this branch! was facing the situation where "`branch is x commits ahead and y commits behind master`" and then I was supposed to force push to overwrite the old changes but I think I used the wrong syntax then it ended up pushing all the commits without overwriting. I apologise for that.. ):

Comment (koonpeng) [reply .storybook/main.js:8]: Seems that a bunch of commits after f0c7931 are duplicated, im not sure what causes that. @matiasbavera Do you have a suggestion to clean it up? I'm thinking of cherry-picking but I don't know if there is a better way.

Comment (koonpeng) [reply .storybook/main.js:8]: I pushed a temp branch `tmp/commands-panel`, @vallq please check if it contains all the newer changes.

Comment (vallq) [reply .storybook/main.js:8]: okay @koonpeng looking at it now!

Comment (matiasbavera) [reply .storybook/main.js:8]: Yeah @koonpeng, I think cherry pick will be our best option here if we want a clean history. But I'm case we don't we can also checkout files from master

`git checkout master --  file.ts`

(Sorry I'm writing from my cellphone)

Comment (matiasbavera) [reply .storybook/main.js:8]: Don't worry @vallq, people "learn by doing" and we are here to help =)

File: src/components/tests/loop-form.test.tsx [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+
+const buildWrapper = (fleetName: string, onClick: any ) => {
+  const wrapper = mount(<LoopForm requestLoop={onClick} fleetNames={[fleetName]} />);
+  return wrapper;
+};
+
+describe('form Validation', () => {
+  afterEach(() => {
+    isRequestButtonClicked = false;
+  });
```
Comment (matiasbavera) [inline src/components/tests/loop-form.test.tsx:26]: Just food for thought, if in the future I create a test outside of this `describe` and if that new test changes the value `isRequestButtonClicked  =  true` and later a test inside the `describe` is executed (let's say 'Number of loops cannot be empty'), I think that 'Number of loops cannot be empty' could fail because it's expecting for `isRequestButtonClicked` to be `false`.

Thinking in that edge case, wouldn't it be better to use `beforeEach` instead of `afterEach`? 


It's also possible that I'm missing something =)

File: e2e/tests/loopRequest.test.ts [hunk 14]
```diff
@@ -12,17 +12,31 @@ describe('loop request', () => {
...
+    backButton.click();
+    $('[data-component=MainMenu] [data-item=Commands]').click();
+    $('input[name=numLoops]').setValue(1);
+    $('button=Request').click();
+
+    backButton.click();
+    $('[data-component=MainMenu] [data-item=Robots]').click();
+    const newRobotLocations = getRobotLocations();
+
+    expect(newRobotLocations).not.toMatchObject(currentRobotLocations);
```
Comment (matiasbavera) [inline e2e/tests/loopRequest.test.ts:40]: I'm getting this error on the e2e test. It's opening the robot accordion instead of the command one.

```
ERROR webdriver: Request failed with status 400 due to element not interactable: element not interactable
  (Session info: chrome=83.0.4103.116)
[0-0] element not interactable in "loop request rmf responds to loop request"
element not interactable
  (Session info: chrome=83.0.4103.116)
```

**Comment by koonpeng** [comment]
For the styling issues, should we use husky to do a prettier run on commit?

File: src/components/tests/loop-form.test.tsx (hunk 13 above)
Comment (koonpeng) [reply src/components/tests/loop-form.test.tsx:26]: I agree, setting the pre-conditions in `beforeEach` is better than setting it in `afterEach` for the next test. Unless there is another use for the var that I'm missing.

To improve it more, we can also move the var inside the `describe` block.

File: e2e/tests/loopRequest.test.ts (hunk 14 above)
Comment (koonpeng) [reply e2e/tests/loopRequest.test.ts:40]: I got a similar error as well, in my case, it seems like the loop request is in an accordion but the test is not updated to click on it.

Comment (vallq) [reply e2e/tests/loopRequest.test.ts:40]: Okay let me get to this, my oversight!!

File: src/components/tests/loop-form.test.tsx (hunk 13 above)
Comment (vallq) [reply src/components/tests/loop-form.test.tsx:26]: you mean moving the initialisation of `isRequestButtonClicked` to inside the describe block?

Comment (koonpeng) [reply src/components/tests/loop-form.test.tsx:26]: yea,

```js
describe('something', () => {
  let foo = false;
});
```

**Comment by vallq** [comment]
I added husky to run prettier formatting on pre-commits! but just checking, are we having new lines at the end of every file?

File: package.json [hunk 15]
```diff
@@ -54,6 +54,16 @@
   "eslintConfig": {
     "extends": "react-app"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.+(js|jsx|json|css|md)": [
```
Comment (koonpeng) [inline package.json:63]: let's revert this and tackle it in a standalone PR.

also ts and tsx is missing, and won't this pattern match repeated patterns like `something.jsjs`?

Comment (matiasbavera) [reply package.json:63]: Yeah, you're right if the library it's using regex. Good catch!

image
image
https://regex101.com/

**Comment by matiasbavera** [comment]
> I added husky to run prettier formatting on pre-commits! but just checking, are we having new lines at the end of every file?

Yeah, I think we should have it. 

Interesting links: 
* https://thoughtbot.com/blog/no-newline-at-end-of-file
* https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file

File: package.json (hunk 15 above)
Comment (vallq) [reply package.json]: oh ok, sorry about that because I followed a tutorial. I've reverted the changes, so we can address this in a new PR

**Comment by vallq** [comment]
the other panels that use the `ExpansionPanel`, `ExpansionSummary` etc. components need to be updated to `Accordion`, `AccordionSummary` as well. else there will be a Material UI error being logged to the console

**Comment by matiasbavera** [comment]
> the other panels that use the `ExpansionPanel`, `ExpansionSummary` etc. components need to be updated to `Accordion`, `AccordionSummary` as well. else there will be a Material UI error being logged to the console

I think we can do this in another PR.

**Comment by matiasbavera** [comment]
Can the form be in the center of the container? Maybe you can check the padding.
image

File: package.json (hunk 15 above)
Comment (matiasbavera) [reply package.json]: I created an Issue about Husky https://github.com/osrf/romi-dashboard/issues/68

**Comment by vallq** [comment]
loop form
How does this look?

**Comment by vallq** [comment]
No issues with the different zoom levels (: 
@koonpeng can it be merged?

**Comment by koonpeng** [comment]
e2e tests are failing for me

```
Starting ChromeDriver 83.0.4103.39 (ccbf011cb2d2b19b506d844400483861342c20cd-refs/branch-heads/4103@{#416}) on port 9515
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
[0-0] RUNNING in chrome - /e2e/tests/loopRequest.test.ts
[1595925765.859][WARNING]: This version of ChromeDriver has not been tested with Chrome version 84.
[0-0] 2020-07-28T08:43:02.263Z WARN webdriver: Request failed with status 400 due to element not interactable
  (Session info: chrome=84.0.4147.105)
[0-0] 2020-07-28T08:43:02.276Z WARN webdriver: Request failed with status 400 due to element not interactable
  (Session info: chrome=84.0.4147.105)
[0-0] 2020-07-28T08:43:02.287Z WARN webdriver: Request failed with status 400 due to element not interactable
  (Session info: chrome=84.0.4147.105)
[0-0] 2020-07-28T08:43:02.299Z ERROR webdriver: Request failed with status 400 due to element not interactable: element not interactable
  (Session info: chrome=84.0.4147.105)
[0-0] element not interactable in "loop request rmf responds to loop request"
element not interactable
  (Session info: chrome=84.0.4147.105)
[0-0] FAILED in chrome - /e2e/tests/loopRequest.test.ts

 "spec" Reporter:
------------------------------------------------------------------
[chrome 84.0.4147.105 linux #0-0] Spec: /home/teokp/rmf/src/romi-dashboard/e2e/tests/loopRequest.test.ts
[chrome 84.0.4147.105 linux #0-0] Running: chrome (v84.0.4147.105) on linux
[chrome 84.0.4147.105 linux #0-0] Session ID: 6374917b5f2d3ef712618b50e466910e
[chrome 84.0.4147.105 linux #0-0]
[chrome 84.0.4147.105 linux #0-0] loop request
[chrome 84.0.4147.105 linux #0-0]    ✖ rmf responds to loop request
[chrome 84.0.4147.105 linux #0-0]
[chrome 84.0.4147.105 linux #0-0] 1 failing (21.8s)
[chrome 84.0.4147.105 linux #0-0]
[chrome 84.0.4147.105 linux #0-0] 1) loop request rmf responds to loop request
[chrome 84.0.4147.105 linux #0-0] element not interactable
  (Session info: chrome=84.0.4147.105)
[chrome 84.0.4147.105 linux #0-0] element not interactable
[chrome 84.0.4147.105 linux #0-0]   (Session info: chrome=84.0.4147.105)
[chrome 84.0.4147.105 linux #0-0]     at processTicksAndRejections (internal/process/task_queues.js:97:5)
[chrome 84.0.4147.105 linux #0-0]     at Context.<anonymous> (/home/teokp/rmf/src/romi-dashboard/e2e/tests/loopRequest.test.ts:34:31)
```

**Comment by vallq** [comment]
is it consistently failing? cos it works fine for me, the e2e tests are passing

**Comment by matiasbavera** [comment]
That's weird it works for me too
```
 "spec" Reporter:
------------------------------------------------------------------
[chrome 84.0.4147.89 linux #0-0] Spec: /home/ekumen/git/osrf/romi-dashboard/e2e/tests/loopRequest.test.ts
[chrome 84.0.4147.89 linux #0-0] Running: chrome (v84.0.4147.89) on linux
[chrome 84.0.4147.89 linux #0-0] Session ID: 3376971213e34e186e4d79f4f5913da4
[chrome 84.0.4147.89 linux #0-0]
[chrome 84.0.4147.89 linux #0-0] loop request
[chrome 84.0.4147.89 linux #0-0]    ✓ rmf responds to loop request
[chrome 84.0.4147.89 linux #0-0]
[chrome 84.0.4147.89 linux #0-0] 1 passing (30.3s)


Spec Files:      1 passed, 1 total (100% completed) in 00:00:32 

Done in 57.81s.
```

**Comment by koonpeng** [comment]
it works after added `waitForClickable`, not sure why, maybe because of animations.
